usb/imx-usbc/controller.c

```diff
@@ -297,7 +327,7 @@ int ctrl_hfIrq(void)
 	int endpt = 0;
 
 	if ((ctrl_common.dc->setupstat = *(ctrl_common.dc->base + endptsetupstat)) & 0x1) {
-		/* trip winre set */
+		/* Trip wire set */
 		while (!((ctrl_common.dc->setupstat >> endpt) & 1))
 			endpt++;
```

Comment:
Usually host is responsible for initiating transmission with the device. So that, there is a check in line 299 whether setup transaction is received on control endpoint and then other endpoints are checked in the while loop.
I think that there is very little chance of that interrupts is received and no setup packet is transmitted on any of the endpoints.

However, if we ended up with endpt > 15, there would be definitely a bad memory access. For this reason it would be safer to check endpt value before using it.

Reply:
Oops. You're right, it could go crazy.

Reply:
Done, also added USB interrupt clear routine.